File: 4.Algorithm/马士兵/algorithm_Java/src/maoyu/simple_sort/Selection_Sort.java
package maoyu.simple_sort;

import java.util.Arrays;

public class Selection_Sort {

    /**
     * 对整型数组进行选择排序（升序）
     *
     * @param arr 待排序的数组
     */
    public void sort(int[] arr) {
        int n = arr.length;
        // 遍历数组中的每一个元素，最后一个元素无需遍历
        for (int i = 0; i < n - 1; i++) {
            // 假设当前索引的元素是未排序部分的最小值
            int minIndex = i;

            // 在未排序部分中查找最小元素的索引
            for (int j = i + 1; j < n; j++){
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            // 将最小元素与当前索引的元素交换位置
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }

    // 示例
    public static void main(String[] args) {
        Selection_Sort sorter = new Selection_Sort();
        int[] myList = {64, 25, 12, 22, 11, 0};
        sorter.sort(myList);
        System.out.println("排序后的数组: " + Arrays.toString(myList));
    }
}

File: 4.Algorithm/马士兵/algorithm_Python/Simple_Sort/Selection_Sort.py
def selection_sort(arr):
    """
        对列表进行选择排序（升序）

        :param arr: 待排序的列表
    """
    # 遍历列表中的每一个元素
    for i in range(len(arr) - 1):
        # 假设当前位置的元素是未排序部分的最小值
        min_index = i

        # 遍历未排序部分，寻找真正的最小值
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[min_index]:
                min_index = j

        # 如果找到了一个更小的元素，则将其与当前位置的元素交换
        if min_index != i:
            arr[i], arr[min_index] = arr[min_index], arr[i]


# 示例
if __name__ == "__main__":
    my_list = [64, 25, 12, 22, 11, 0, 99]
    selection_sort(my_list)
    print("排序后的数组:", my_list)